### PR TITLE
Fix misuse of layoutMode prop in desktop view

### PR DESCRIPTION
**Pull request by GustavoMA**
#### What is the purpose of this pull request?
As described in the title.

#### How should this be manually tested?
[Here](https://alves--phmex.myvtex.com/)

#### Types of changes

* [x] Bug fix (a non-breaking change which fixes an issue)
* [ ] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/SearchResult.js [hunk 1]
```diff
@@ -39,15 +39,15 @@ class SearchResult extends Component {
     hiddenFacets: this.props.hiddenFacets,
   }
 
-  handleLayoutChange = e => {
+  handleMobileLayoutChange = e => {
     e.preventDefault()
 
-    const defaultModes = [this.props.hiddenFacets.layoutMode1, this.props.hiddenFacets.layoutMode2]
-    const modeIndex = (defaultModes.indexOf(this.state.galleryLayoutMode) + 1) % 2
+    const defaultModes = [this.props.mobileLayout.mode1, this.props.mobileLayout.mode2]
```
Comment (rerissondaniel) [inline react/components/SearchResult.js]: I'm not sure if defaultModes makes sense here, maybe only 'modes' would be more descriptive and avoid misunderstandings

File: react/index.js [hunk 2]
```diff
@@ -85,27 +85,34 @@ SearchResultQueryLoader.getSchema = props => {
     type: 'object',
     properties: {
       ...querySchema,
-      hiddenFacets: {
-        title: 'editor.search-result.hiddenFacets',
+      mobileLayout: {
+        title: 'Mobile Layout Switcher',
```
Comment (rerissondaniel) [inline react/index.js]: Ids should be used in order to support internationalization

File: react/components/SearchResult.js (hunk 1 above)
Comment (GustavoMA) [reply react/components/SearchResult.js]: Fixed.

File: react/Gallery.js [hunk 3]
```diff
@@ -54,8 +59,8 @@ Gallery.propTypes = {
   products: PropTypes.arrayOf(productShape),
   /** ProductSummary props. */
   summary: PropTypes.any,
-  /** Layout mode of the gallery */
-  layoutMode: PropTypes.string,
+  /** Layout mode of the gallery in mobile view */
+  mobileLayoutMode: PropTypes.string,
```
Comment (luciannojunior) [inline react/Gallery.js]: Are the layout modes discrete? You could use PropTypes' `oneOf' to make it more clear. Maybe add some constantes on the top of the page and use it in the conditionals...

Comment (luciannojunior) [reply react/Gallery.js]: There are some proptypes that are not required and algo they are not on defaultProps. Pls check

File: react/components/SearchResult.js [hunk 4]
```diff
@@ -15,14 +15,14 @@ class SearchResult extends Component {
   static propTypes = searchResultPropTypes
 
   static defaultProps = {
-    hiddenFacets: {
-      layoutMode1: LAYOUT_MODE[0].value,
-      layoutMode2: LAYOUT_MODE[1].value,
+    mobileLayout: {
```
Comment (luciannojunior) [inline react/components/SearchResult.js]: Check if the [propTypes](https://github.com/vtex-apps/search-result/blob/master/react/constants/propTypes.js) are correct

File: react/index.js [hunk 5]
```diff
@@ -85,27 +85,34 @@ SearchResultQueryLoader.getSchema = props => {
...
         type: 'object',
         isLayout: true,
         properties: {
-          layoutMode1: {
-            title: 'Layout Mode Switcher 1',
+          mode1: {
+            title: 'Layout Mode 1',
```
Comment (luciannojunior) [inline react/index.js]: Titles here could be internationalized

File: react/index.js (hunk 2 above)
Comment (GustavoMA) [reply react/index.js]: Fixed.

File: react/index.js (hunk 5 above)
Comment (GustavoMA) [reply react/index.js]: Fixed.

File: react/Gallery.js (hunk 3 above)
Comment (GustavoMA) [reply react/Gallery.js]: Fixed.

File: react/components/SearchResult.js [hunk 6]
```diff
@@ -15,14 +15,14 @@ class SearchResult extends Component {
...
+      mode1: LAYOUT_MODE[0].value,
+      mode2: LAYOUT_MODE[1].value,
     },
   }
 
   state = {
-    galleryLayoutMode: this.props.hiddenFacets.layoutMode1 || LAYOUT_MODE[0].value,
+    mobileLayoutMode: this.props.mobileLayout.mode1,
```
Comment (theomoura) [inline react/components/SearchResult.js]: Please, double check this change, it was not working properly only with the props before.

Comment (GustavoMA) [reply react/components/SearchResult.js]: `this.props.mobileLayout.mode1` is already in the `defaultProps`.
But I also tested it out and it is working properly.

File: react/components/SearchResult.js (hunk 4 above)
Comment (GustavoMA) [reply react/components/SearchResult.js]: Fixed.

**Pull request merged**